data/trie/sync.go

```diff
@@ -100,11 +100,6 @@ func (ts *trieSyncer) StartSyncing(rootHash []byte, ctx context.Context) error {
 
 		numUnResolved := ts.requestNodes()
 		if !shouldRetryAfterRequest && numUnResolved == 0 {
-			err = ts.trie.Commit()
```

Comment:
why delete this ? isn't it better to have the final commit done ?

Reply:
If we commit here, the whole trie will be loaded again from storage. The current approach is to commit each node as it is synced.